src/utilities/tab-delimited-to-json-converter-factory.js

```diff
@@ -0,0 +1,208 @@
+var libsbgnjs = require('libsbgn.js');
+var parseString = require('xml2js').parseString;
+var libUtilities = require('./lib-utilities');
+var classes = require('./classes');
+
+module.exports = function() {
+	var elementUtilities;
+
+	function tdToJson(param) {
+    optionUtilities = param.optionUtilities;
+    options = optionUtilities.getOptions();
+    elementUtilities = param.elementUtilities;
+  	}
+
+  	tdToJson.map = undefined;
+
+	tdToJson.edgeTypes= { 
+		'positive influence': true, 
+		'negative influence': true, 
+		'unknown influence': true, 
+		'necassary influence': true,
+		'logic arcs': true, 
+		'equivalance arcs': true 
+	};
+
+	tdToJson.nodeTypes= {
+		'biological activity': true,
+		'macromolecule': true, 
+		'complex': true, 
+		'simple chemical': true, 
+		'unspecified entity': true, 
+		'nucleic acid feature': true,
+		'perturbing agent':true, 
+		'phenotype': true, 
+		'compartment': true, 
+		'submap': true, 
+		'tag': true, 
+		'and': true, 
+		'or': true, 
+		'not': true, 
+		'delay': true
+	};
+
+	tdToJson.validateNodeType = function( type){
+		var _type = type.replace('_', ' ').toLowerCase();
+		if( this.nodeTypes[_type] == true)
+	 		return true;
+		return false;
+	};
+
+	tdToJson.validateEdgeType = function( type){
+		var _type = type.replace('_', ' ').toLowerCase();
+		if( this.edgeTypes[_type] == true)
+			return true;
+		return false;
+	};
+
+	tdToJson.convertTypeToClass = function( ele, type){
+		ele.data.class =  type.replace('_', ' ').toLowerCase();
```

Comment:
@kaansancak node type from file will not always be the same with node class in SBGNViz/ChiSE. The followings are different:

|Type in file:| Class in SBGNViz|
|----------------|---------------------|
|biological activity | BA plain |
|macromolecule | BA macromolecule|
|complex | BA complex|
|simple chemical | BA chemical|
|unspecified entity | BA unspecified entity|
|nucleic acid feature | BA nucleic acid feature|
|perturbing agent | BA perturbing agent| 

One way of implementing this: set the values of node (edges) types in `tdToJson.nodeTypes` to actual SBGNViz classes and use them here.

Reply:
@kaansancak without setting node class correctly, some features will not work properly.

Reply:
@kinimesi SBGNViz has both "macromolecule" and "BA macromolecule" classes. Should I ignore the macrolemolecule class and set all 'biological activity' as BA plain or should I differentiate two classes?

Reply:
@kaansancak "macromolecule" is the class name for PD genes, whereas "BA macromolecule" is the class name for AF genes. You are creating an AF map in this case.